pywt/multilevel.py

```diff
@@ -220,3 +220,156 @@ def waverec2(coeffs, wavelet, mode='sym'):
+        Wavelet to use
+
+    Examples
+    --------
+    >>> import pywt
+    >>> coeffs = pywt.swt([1,2,3,4,5,6,7,8], 'db2', level=2)
+    >>> pywt.iswt(coeffs, 'db2')
+    array([ 1.,  2.,  3.,  4.,  5.,  6.,  7.,  8.])
+    """
+
+    output = coeffs[0][0].copy()  # Avoid modification of input data
+
+    # num_levels, equivalent to the decomposition level, n
+    num_levels = len(coeffs)
+    for j in range(num_levels,0,-1):
+        step_size = int(pow(2, j-1))
+        last_index = step_size
+        _, cD = coeffs[num_levels - j]
+        for first in range(last_index):  # 0 to last_index - 1
+
+            # Getting the indices that we will transform
+            indices = np.arange(first, len(cD), step_size)
+
+            # select the even indices
+            even_indices = indices[0::2]
+            # select the odd indices
+            odd_indices = indices[1::2]
+
+            # perform the inverse dwt on the selected indices,
+            # making sure to use periodic boundary conditions
+            x1 = idwt(output[even_indices], cD[even_indices], wavelet, 'per')
+            x2 = idwt(output[odd_indices], cD[odd_indices], wavelet, 'per')
+
+            # perform a circular shift right
+            x2 = np.roll(x2, 1)
+
+            # average and insert into the correct indices
+            output[indices] = (x1 + x2)/2.
+
+    return output
+
+
+def iswt2(coeffs, wavelet):
+    """
+    Multilevel 2D Inverse Discrete Stationary Wavelet Transform.
+
+    Parameters
+    ----------
+    coeffs : list
+        Approximation and details coefficients::
+
+            [
+                (cA_1,
+                    (cH_1, cV_1, cD_1)
+                ),
+                (cA_2,
+                    (cH_2, cV_2, cD_2)
+                ),
+                ...,
+                (cA_n
+                    (cH_n, cV_n, cD_n)
+                )
+            ]
+
+        where cA is approximation, cH is horizontal details, cV is
+        vertical details, cD is diagonal details and n is number of
+        levels.
+    wavelet : Wavelet object or name string
+        Wavelet to use
+
+    Examples
+    --------
+    >>> import pywt
+    >>> coeffs = coeffs = pywt.swt2([[1,2,3,4],[5,6,7,8],
+                                     [9,10,11,12],[13,14,15,16]],
+                                    'db1', level=2)
+    >>> pywt.iswt2(coeffs, 'db1')
+    array([[  1.,   2.,   3.,   4.],
+           [  5.,   6.,   7.,   8.],
+           [  9.,  10.,  11.,  12.],
+           [ 13.,  14.,  15.,  16.]])
+
+    """
+
+    output = coeffs[-1][0].copy()  # Avoid modification of input data
+
+    # num_levels, equivalent to the decomposition level, n
+    num_levels = len(coeffs)
+    for j in range(num_levels,0,-1):
+        step_size = int(pow(2, j-1))
+        last_index = step_size
+        _, (cH, cV, cD) = coeffs[j-1]
+        # We are going to assume cH, cV, and cD are square and of equal size
+        assert(cH.shape == cV.shape)
```

Comment:
Plain ``assert``s get stripped from the code when byte-compiling with ``-O``, so they shouldn't be used at all. In tests you want ``np.testing.assert_`` instead, here I think something like this would be cleaner:

    if cH.shape != cV.shape:
        raise RuntimeError("Mismatch in shape of intermediate coefficient arrays")